### PR TITLE
Add support for TypeScript

### PR DESCRIPTION
**Pull request by ryanwe**
Explicitly ignore any type declaration files. While not explicitly a problem, this causes issues when running TypeScript via `ts-node`. See https://github.com/TypeStrong/ts-node/pull/207 for more details.

### CODE REVIEW

**Comment by aseemk** [comment]
Thank you! 🙏

**Review by aseemk — APPROVED** [review]

File: test/simple/f.d.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+export const foo: 'bar';
```
Comment (blakeembrey) [inline test/simple/f.d.ts]: A more valid `.d.ts` file wouldn't have values, remember 😄 It'd be `export const foo: string;`

Comment (blakeembrey) [inline test/simple/f.d.ts]: Ignore me, string literal types! Missed the colon. Deleting these!

**Comment by brandonros** [comment]
What about adding an `index.d.ts` file to the root directory of the repo so that it is possible to `import * as requireDir from 'require-dir'`?

**Comment by yocontra** [comment]
@brandonros Haven't seen that done before, don't think that is necessary

**Comment by brandonros** [comment]
It is. Try to `import * as requireDir from 'require-dir'` from any TypeScript project. It will throw an error about a lacking module declaration. You can either add it to the repo like https://github.com/axios/axios/blob/master/index.d.ts or create @types/require-dir (like https://www.npmjs.com/package/@types/express)

**Comment by yocontra** [comment]
@brandonros Yep typings I understand - I would rather go that route. I don't use typescript, but if somebody else wants to do them it should be a pretty simple API to model.

**Pull request merged**